numpyro/infer/util.py

```diff
@@ -164,6 +168,50 @@ def potential_energy(model, inv_transforms, model_args, model_kwargs, params):
     return - log_joint
 
 
+def enum_potential_energy(model, inv_transforms, model_args, model_kwargs, params):
```

Comment:
@fehiepsi @neerajprad here's the modified `potential_energy` we discussed

Reply:
Nice..I think this should plug into the existing hmc implementation without issues. Let us know if you face any issues.

Reply:
I've been trying to run HMC with `enum_potential_energy` but I'm currently a bit stuck trying to work around the assumption baked into multiple parts of the HMC code that every latent variable is differentiable and constrained.  Is there an easy way to hack around this for testing?

Reply:
I think we'll need to change both `get_potential_fn` and `find_valid_initial_params` in `numpyro.infer.util` to exclude sample sites containing discrete valued distributions.